lk2nd/hw/gpio/pm8x41.c

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright (c) 2022 Nikita Travkin <nikita@trvn.ru> */
+
+#include <stdint.h>
+#include <bits.h>
+#include <debug.h>
+#include <pm8x41.h>
+#include <pm8x41_hw.h>
+
+#include <lk2nd/gpio.h>
+#include <lk2nd/util/minmax.h>
+
+#include "supplier.h"
+
+#define PMIC_GPIO_NUM_PIN(num)   BITS_SHIFT(num, 7, 0)
+#define PMIC_GPIO_NUM_SID(num)   BITS_SHIFT(num, 15, 8)
+
+#define PMIC_FLAGS_VIN_SEL(x)	BITS_SHIFT(num, 26, 24)
+#define PMIC_NON_DEFAULT_VIN_SEL	BIT(27)
```

Comment:
I was thinking if we should define all the `BIT()` and `BITS_SHIFT()` in the dt-bindings as well just to have the bit positions at one consistent place. it's fine as long as it's not expanded within DTs